src/Purestone/Language/Program.hs

```diff
@@ -18,6 +21,8 @@ type Program = [Instr]
 -- | Represents instructions that can be used to program cards with.
 data Instr 
     = NoOp 
-    deriving (Eq, Show)
+    deriving (Eq, Show, Read, Generic)
```

Comment:
I don't think the `Read` instance here is overly sensible since we will implement a proper parser for this later

Reply:
Sorry, I got confused, none of the `Read` instances were necessary for this, so I've removed them all.